driver-core/src/main/java/com/datastax/driver/core/SchemaParser.java

```diff
@@ -848,9 +855,344 @@ private boolean supportsUdfs(VersionNumber cassandraVersion) {
+            ExecutionException {
+      String queryPrefix =
+          SELECT_AGGREGATES + whereClause(KEYSPACE, keyspace.getName(), null, null);
+      List<Row> result = new ArrayList<Row>();
+      List<Row> rs = queryAsync(queryPrefix + LIMIT, connection, protocolVersion).get().all();
+      while (!rs.isEmpty()) {
+        String lastSeenAggregate = "'" + rs.get(rs.size() - 1).getString(AGGREGATE_NAME) + "'";
+        String lastSeenArgs = "'" + rs.get(rs.size() - 1).getString(ARGUMENT_TYPES) + "'";
+        result.addAll(rs);
+        rs =
+            queryAsync(
+                    queryPrefix
+                        + " AND ("
+                        + AGGREGATE_NAME
+                        + ", "
+                        + ARGUMENT_TYPES
+                        + ") > ("
+                        + lastSeenAggregate
+                        + ", "
+                        + lastSeenArgs
+                        + ")"
+                        + LIMIT,
+                    connection,
+                    protocolVersion)
+                .get()
+                .all();
+      }
+      return result;
+    }
+
+    private void buildAggregates(
+        KeyspaceMetadata keyspace,
+        Cluster cluster,
+        Connection connection,
+        VersionNumber cassandraVersion,
+        ProtocolVersion protocolVersion)
+        throws ConnectionException, BusyConnectionException, InterruptedException,
+            ExecutionException {
+      List<Row> raw = fetchAggregates(keyspace, connection, protocolVersion);
+      Map<String, AggregateMetadata> aggregates =
+          buildAggregates(keyspace, raw, cassandraVersion, cluster);
+      for (AggregateMetadata aggregate : aggregates.values()) {
+        keyspace.add(aggregate);
+      }
+    }
+
+    private Map<String, Map<String, Row>> fetchColumns(
+        KeyspaceMetadata keyspace, Connection connection, ProtocolVersion protocolVersion)
+        throws ConnectionException, BusyConnectionException, InterruptedException,
+            ExecutionException {
+      String queryPrefix = SELECT_COLUMNS + whereClause(KEYSPACE, keyspace.getName(), null, null);
+      Map<String, Map<String, Row>> result = new HashMap<String, Map<String, Row>>();
+      List<Row> rs = queryAsync(queryPrefix + LIMIT, connection, protocolVersion).get().all();
+      while (!rs.isEmpty()) {
+        String lastSeenTable = "'" + rs.get(rs.size() - 1).getString(TABLE_NAME) + "'";
+        String lastSeenColumn = "'" + rs.get(rs.size() - 1).getString(COLUMN_NAME) + "'";
+        for (Row row : rs) {
+          String cfName = row.getString(TABLE_NAME);
+          Map<String, Row> colsByCf = result.get(cfName);
+          if (colsByCf == null) {
+            colsByCf = new HashMap<String, Row>();
+            result.put(cfName, colsByCf);
+          }
+          colsByCf.put(row.getString(ColumnMetadata.COLUMN_NAME), row);
+        }
+        rs =
```

Comment:
Just being a little paranoid: Is it OK to execute those `SELECT_COLUMNS` queries back-to-back without any delays in-between? Although a temporary query spike shouldn't be a problem?

Reply:
Previously all this data was fetched with just a few parallel unpaged queries so I think it's only better now when we're fetching limited amount of data sequentially.